### ISSUE TITLE
Cannot find module 'libs/features/ui/base', ReferenceError: __importDefault is not defined

### ISSUE THREAD

**Issue by curiouscod3** [nstudio/xplat#186]
I'm using latest NX version. Installing with xplat was successful.. but
**ng serve ionic-my gives this error**

Even though I've changed the HeaderComponent and HeaderBaseComponent import path, it still doesn't recognize the path when ng serve. 

> "dependencies": {
>     "@angular/animations": "9.0.0",
>     "@angular/common": "9.0.0",
>     "@angular/compiler": "9.0.0",
>     "@angular/core": "9.0.0",
>     "@angular/forms": "9.0.0",
>     "@angular/platform-browser": "9.0.0",
>     "@angular/platform-browser-dynamic": "9.0.0",
>     "@angular/platform-server": "9.0.0",
>     "@angular/router": "9.0.0",
>     "@nestjs/common": "^6.8.3",
>     "@nestjs/core": "^6.8.3",
>     "@nestjs/ng-universal": "^3.1.0",
>     "@nestjs/platform-express": "^6.8.3",
>     "@nguniversal/express-engine": "^9.0.2",
>     "@nrwl/angular": "9.1.4",
>     "@nstudio/xplat": "^9.0.1",
>     "class-transformer": "^0.2.3",
>     "class-validator": "^0.9.1",
>     "core-js": "^2.5.4",
>     "express": "^4.15.2",
>     "reflect-metadata": "^0.1.12",
>     "rxjs": "~6.5.0",
>     "zone.js": "^0.10.2",
>     "@ionic-native/core": "^5.10.0",
>     "@ionic-native/splash-screen": "^5.10.0",
>     "@ionic-native/status-bar": "^5.10.0",
>     "@ionic/angular": "^4.7.0",
>     "@fullstack-x/scss": "file:libs/scss",
>     "@ngx-translate/core": "~11.0.0",
>     "@ngx-translate/http-loader": "~4.0.0",
>     "@fullstack-x/web-scss": "file:xplat/web/scss",
>     "@fullstack-x/ionic-scss": "file:xplat/ionic/scss"
>   },
>   "devDependencies": {
>     "@angular-devkit/architect": "^0.803.3",
>     "@angular-devkit/build-angular": "^0.803.3",
>     "@angular-devkit/core": "8.3.3",
>     "@angular-devkit/schematics": "8.3.3",
>     "@angular/cli": "9.0.1",
>     "@angular/compiler-cli": "9.0.0",
>     "@angular/language-service": "9.0.0",
>     "@capacitor/cli": "~1.1.1",
>     "@capacitor/core": "^1.5.2",
>     "@ionic/angular-toolkit": "^2.0.0",
>     "@nestjs/schematics": "^6.7.0",
>     "@nestjs/testing": "^6.8.3",
>     "@nguniversal/builders": "^9.0.2",
>     "@nrwl/angular": "9.1.4",
>     "@nrwl/cypress": "9.1.4",
>     "@nrwl/jest": "9.1.4",
>     "@nrwl/nest": "9.1.4",
>     "@nrwl/node": "9.1.4",
>     "@nrwl/workspace": "9.1.4",
>     "@nstudio/angular": "9.0.1",
>     "@nstudio/ionic": "9.0.1",
>     "@nstudio/ionic-angular": "9.0.1",
>     "@nstudio/web-angular": "9.0.1",
>     "@nstudio/xplat": "9.0.1",
>     "@types/express": "^4.17.0",
>     "@types/jasmine": "~2.8.8",
>     "@types/jasminewd2": "~2.0.3",
>     "@types/jest": "~25.1.4",
>     "@types/node": "~8.9.4",
>     "codelyzer": "~4.5.0",
>     "cypress": "^4.1.0",
>     "dotenv": "6.2.0",
>     "eslint": "6.8.0",
>     "jasmine-core": "~2.99.1",
>     "jasmine-spec-reporter": "~4.2.1",
>     "jest": "24.9.0",
>     "jest-jasmine2": "~25.2.0",
>     "jest-preset-angular": "~8.1.2",
>     "karma": "~4.1.0",
>     "karma-chrome-launcher": "~2.2.0",
>     "karma-coverage-istanbul-reporter": "~2.0.1",
>     "karma-jasmine": "~1.1.2",
>     "karma-jasmine-html-reporter": "^0.2.2",
>     "prettier": "1.19.1",
>     "protractor": "^5.4.2",
>     "ts-jest": "24.0.0",
>     "ts-node": "~7.0.0",
>     "tslint": "~6.0.0",
>     "typescript": "~3.7.4"
>   },

```
Uncaught ReferenceError: __importDefault is not defined
    at Module.../../xplat/web/features/ui/components/header/header.component.ts (header.component.ts:7)
    at __webpack_require__ (bootstrap:84)
    at Module.../../xplat/web/features/ui/components/index.ts (header.component.ts:9)
    at __webpack_require__ (bootstrap:84)
    at Module.../../xplat/web/features/ui/index.ts (index.ts:1)
    at __webpack_require__ (bootstrap:84)
    at Module.../../xplat/web/features/index.ts (index.ts:2)
    at __webpack_require__ (bootstrap:84)
    at Module.../../xplat/web/index.ts (ui.module.ts:28)
    at __webpack_require__ (bootstrap:84)
```

```
ERROR in ../../libs/features/ui/base/header.base-component.ts:4:31 - error TS2307: Cannot find module 'libs/core/base'.

4 import { BaseComponent } from 'libs/core/base';
                                ~~~~~~~~~~~~~~~~
../../xplat/web/features/ui/components/header/header.component.ts:2:37 - error TS2307: Cannot find module 'libs/features/ui/base'.

2 import { HeaderBaseComponent } from 'libs/features/ui/base';
                                      ~~~~~~~~~~~~~~~~~~~~~~~
```

I've checked pre-installed features and other modules related to xplat. 
A lot of things don't import modules properly  such as UIModules stuff like that.

Is it required to use the features lib for xplat?? 

**how to use "--ignoreBase" ?**

**Comment by phillipplum** [comment]
I have this problem to and i think it is based on this issue: https://github.com/angular/angular/issues/32215. It is the mix of typescript and angular version. When i upgrade to angular 9 and ionic 5 everything is ok. Look at this comment: https://github.com/nstudio/xplat/issues/180#issuecomment-604968547

**Comment by curiouscod3** [comment]
> I have this problem to and i think it is based on this issue: [angular/angular#32215](https://github.com/angular/angular/issues/32215). It is the mix of typescript and angular version. When i upgrade to angular 9 and ionic 5 everything is ok. Look at this comment: [#180 (comment)](https://github.com/nstudio/xplat/issues/180#issuecomment-604968547)

I've clean installed with Angular 9 and Ionic 5; look at dependencies in my package.json 

I am asking why this library forces to install the base UI theme?. I'd like to clean install with it. it is totally mass in them

**Comment by johannesschobel** [comment]
Dear @curiouscod3 , please check my issue https://github.com/nstudio/xplat/issues/180 and my linked repository here https://github.com/johannesschobel/nrwl-xplat-ionic5-example . This repo provides a clean installation instruction.

I noticed this issue as well - it is due to the fact that **not all of your dependencies match the same build numbers**. For me it was, that the `capacitor` and `ionic` versions were not up-to-date, i guess.

Please update your `package.json` to reflect the latest dependencies and try again!

**Comment by curiouscod3** [comment]
Dear @johannesschobel  I'ill check it out and report here :) My ng cli version was up-to-date. NX installed the versions in package.json by default. NX was up-to-date as well.

**Comment by curiouscod3** [comment]
@johannesschobel I've clean installed again using using "npx create-nx-workspace@latest".

My NX,Angular CLI and other things(also upgraded Node from 10.x LTS to 12.x LTS) are all up-to-date.

Looks at my package.json that npx create-nx-workspace@latest installed.
Your package.json in nrwl-xplat-ionic5-example is different with mine.
So I guess you manually upgrade each packges. 

```
"dependencies": {
    "@nrwl/angular": "9.1.4",
    "@angular/animations": "9.0.0",
    "@angular/common": "9.0.0",
    "@angular/compiler": "9.0.0",
    "@angular/core": "9.0.0",
    "@angular/forms": "9.0.0",
    "@angular/platform-browser": "9.0.0",
    "@angular/platform-browser-dynamic": "9.0.0",
    "@angular/router": "9.0.0",
    "core-js": "^2.5.4",
    "rxjs": "~6.5.0",
    "zone.js": "^0.10.2",
    "@nestjs/common": "^6.8.3",
    "@nestjs/core": "^6.8.3",
    "@nestjs/platform-express": "^6.8.3",
    "reflect-metadata": "^0.1.13"
  },
  "devDependencies": {
    "@angular/cli": "9.0.1",
    "@nrwl/workspace": "9.1.4",
    "@types/node": "~8.9.4",
    "dotenv": "6.2.0",
    "ts-node": "~7.0.0",
    "tslint": "~6.0.0",
    "eslint": "6.8.0",
    "typescript": "~3.7.4",
    "prettier": "1.19.1",
    "@nrwl/nest": "9.1.4",
    "@angular/compiler-cli": "9.0.0",
    "@angular/language-service": "9.0.0",
    "@angular-devkit/build-angular": "0.900.1",
    "codelyzer": "~5.0.1",
    "jest-preset-angular": "8.0.0",
    "@nrwl/jest": "9.1.4",
    "jest": "24.9.0",
    "@types/jest": "24.9.1",
    "ts-jest": "24.0.0",
    "cypress": "^4.1.0",
    "@nrwl/cypress": "9.1.4",
    "@nrwl/node": "9.1.4",
    "@nestjs/schematics": "^6.7.0",
    "@nestjs/testing": "^6.8.3"
  }
```

I wrote this issue on nx as well before.
https://github.com/nrwl/nx/issues/2751

**Comment by curiouscod3** [comment]
ng generate app with supporting architecture: YES
```
Uncaught ReferenceError: __importDefault is not defined
    at Module.../../xplat/web/features/ui/components/header/header.component.ts (header.component.ts:7)
    at __webpack_require__ (bootstrap:84)
    at Module.../../xplat/web/features/ui/components/index.ts (header.component.ts:9)
    at __webpack_require__ (bootstrap:84)
    at Module.../../xplat/web/features/ui/index.ts (index.ts:1)
    at __webpack_require__ (bootstrap:84)
    at Module.../../xplat/web/features/index.ts (index.ts:2)
    at __webpack_require__ (bootstrap:84)
    at Module.../../xplat/web/index.ts (ui.module.ts:28)
    at __webpack_require__ (bootstrap:84)
```

**ng generate app --useXplat=false**
```

Uncaught ReferenceError: __importDefault is not defined
    at Module../src/app/app.component.ts (VM39 main.js:586)
    at __webpack_require__ (VM37 runtime.js:85)
    at Module../src/app/app.module.ts (VM39 main.js:615)
    at __webpack_require__ (VM37 runtime.js:85)
    at Module../src/main.ts (VM39 main.js:688)
    at __webpack_require__ (VM37 runtime.js:85)
    at Object.0 (VM39 main.js:715)
    at __webpack_require__ (VM37 runtime.js:85)
    at checkDeferredModules (VM37 runtime.js:46)
    at Array.webpackJsonpCallback [as push] (VM37 runtime.js:33)
```

**Comment by curiouscod3** [comment]
@johannesschobel  @NathanWalker phillipplum
I finally made angular9+nest7+ionic5/capacitor mono repo with Xplat.

The issue was from xplat schematics(ng add). It changed everything from v9 to v8.
**So, I had to "ng update --all --force --allowDirty"**
First time, it didn't work and still got the error : "__importDefault is not defined"

**The key was  to not add xplat architecture supporting.** 
In my repo, so there is no xplat folder anymore.
So, keep that it mind. xplat is unstable for now. 
**ng add @nstudio/xplat and choose not right now or it generates xplat anyway.**

**Whenever "ng generate @nstudio/xplat:app", it changes angular version to v8. this was the problem.** 
After a lot of time, I got this mono repo.
I will upload this project on my github soon. 

```
.
├── apps
│   ├── api
│   │   └── src
│   │       ├── app
│   │       ├── assets
│   │       └── environments
│   ├── fullstack-nx-xplat-ng9
│   │   └── src
│   │       ├── app
│   │       ├── assets
│   │       └── environments
│   ├── fullstack-nx-xplat-ng9-e2e
│   │   └── src
│   │       ├── fixtures
│   │       ├── integration
│   │       ├── plugins
│   │       └── support
│   └── ionic-mobile
│       ├── e2e
│       │   └── src
│       └── src
│           ├── app
│           │   └── home
│           ├── assets
│           │   └── icon
│           ├── environments
│           └── theme
├── libs
│   └── api-interfaces
│       └── src
│           └── lib
└── tools
    └── schematics

```

```
{
  "name": "fullstack-nx-xplat-ng9",
  "version": "0.0.0",
  "license": "MIT",
  "scripts": {
    "ng": "ng",
    "nx": "nx",
    "start": "ng serve",
    "build": "ng build",
    "test": "ng test",
    "lint": "nx workspace-lint && ng lint",
    "e2e": "ng e2e",
    "affected:apps": "nx affected:apps",
    "affected:libs": "nx affected:libs",
    "affected:build": "nx affected:build",
    "affected:e2e": "nx affected:e2e",
    "affected:test": "nx affected:test",
    "affected:lint": "nx affected:lint",
    "affected:dep-graph": "nx affected:dep-graph",
    "affected": "nx affected",
    "format": "nx format:write",
    "format:write": "nx format:write",
    "format:check": "nx format:check",
    "update": "ng update @nrwl/workspace",
    "workspace-schematic": "nx workspace-schematic",
    "dep-graph": "nx dep-graph",
    "help": "nx help",
    "postinstall": "ngcc --properties es2015 browser module main --first-only --create-ivy-entry-points",
    "clean": "npx rimraf -- hooks node_modules package-lock.json && npm i",
    "build.ionic.mobile": "nx build ionic-mobile",
    "prepare.ionic.mobile": "npm run clean && npm run clean.ionic.mobile && npm run build.ionic.mobile",
    "prepare.ionic.mobile.ios": "npm run prepare.ionic.mobile && cd apps/ionic-mobile && npm run cap.add.ios",
    "prepare.ionic.mobile.android": "npm run prepare.ionic.mobile && cd apps/ionic-mobile && npm run cap.add.android",
    "open.ionic.mobile.ios": "cd apps/ionic-mobile && npm run cap.ios",
    "open.ionic.mobile.android": "cd apps/ionic-mobile && npm run cap.android",
    "sync.ionic.mobile": "cd apps/ionic-mobile && npm run cap.copy",
    "clean.ionic.mobile": "cd apps/ionic-mobile && npx rimraf -- hooks node_modules platforms www plugins ios android package-lock.json && npm i && rimraf -- package-lock.json"
  },
  "private": true,
  "dependencies": {
    "@angular/animations": "9.1.0",
    "@angular/common": "9.1.0",
    "@angular/compiler": "9.1.0",
    "@angular/core": "9.1.0",
    "@angular/forms": "9.1.0",
    "@angular/platform-browser": "9.1.0",
    "@angular/platform-browser-dynamic": "9.1.0",
    "@angular/router": "9.1.0",
    "@nestjs/common": "^7.0.7",
    "@nestjs/core": "^7.0.7",
    "@nestjs/platform-express": "^7.0.7",
    "@nrwl/angular": "9.1.4",
    "@nstudio/xplat": "^9.0.1",
    "core-js": "^3.6.4",
    "reflect-metadata": "^0.1.12",
    "rxjs": "~6.5.4",
    "zone.js": "^0.10.2",
    "@ionic-native/core": "^5.10.0",
    "@ionic-native/splash-screen": "^5.10.0",
    "@ionic-native/status-bar": "^5.10.0",
    "@ionic/angular": "^5.0.7"
  },
  "devDependencies": {
    "@angular-devkit/architect": "^0.901.0",
    "@angular-devkit/build-angular": "^0.901.0",
    "@angular-devkit/core": "9.1.0",
    "@angular-devkit/schematics": "9.1.0",
    "@angular/cli": "9.1.0",
    "@angular/compiler-cli": "9.1.0",
    "@angular/language-service": "9.1.0",
    "@capacitor/cli": "~1.5.2",
    "@capacitor/core": "^1.5.2",
    "@ionic/angular-toolkit": "^2.0.0",
    "@nestjs/schematics": "^7.0.0",
    "@nestjs/testing": "^7.0.7",
    "@nrwl/angular": "9.1.4",
    "@nrwl/cypress": "9.1.4",
    "@nrwl/jest": "9.1.4",
    "@nrwl/nest": "9.1.4",
    "@nrwl/node": "9.1.4",
    "@nrwl/workspace": "9.1.4",
    "@nstudio/angular": "9.0.1",
    "@nstudio/ionic": "9.0.1",
    "@nstudio/ionic-angular": "9.0.1",
    "@nstudio/web-angular": "9.0.1",
    "@types/jasmine": "~3.5.10",
    "@types/jasminewd2": "~2.0.3",
    "@types/jest": "25.1.4",
    "@types/node": "~13.9.5",
    "codelyzer": "~5.2.2",
    "cypress": "^4.1.0",
    "dotenv": "8.2.0",
    "eslint": "6.8.0",
    "jasmine-core": "~3.5.0",
    "jasmine-spec-reporter": "~5.0.1",
    "jest": "25.2.4",
    "jest-preset-angular": "8.1.2",
    "karma": "~4.4.1",
    "karma-chrome-launcher": "~3.1.0",
    "karma-coverage-istanbul-reporter": "~2.1.1",
    "karma-jasmine": "~3.1.1",
    "karma-jasmine-html-reporter": "^1.5.3",
    "prettier": "2.0.2",
    "protractor": "^5.4.2",
    "ts-jest": "25.3.0",
    "ts-node": "~8.8.1",
    "tslint": "~6.1.0",
    "typescript": "~3.8.3"
  },
  "xplat": {
    "prefix": "fullstack-nx-xplat-ng9",
    "framework": "angular"
  }
}

```

**Comment by NathanWalker** [comment]
This is resolved with new `9.0.2` workspaces now 👍
